econml/sklearn_extensions/linear_model.py

```diff
@@ -1905,3 +1905,130 @@ def model_gen(y):
+
+    def __init__(self, cov_type="HC0"):
+        self.fit_intercept = False
+        self.cov_type = cov_type
+        return
+
+    def _check_input(self, Z, T, y, sample_weight):
+        """Check dimensions and other assertions."""
+
+        # set default values for None
+        if sample_weight is None:
+            sample_weight = np.ones(y.shape[0])
+
+        # check array shape
+        assert (T.shape[0] == Z.shape[0] == y.shape[0] == sample_weight.shape[0]), "Input lengths not compatible!"
+
+        # weight X and y
+        weighted_Z = Z * np.sqrt(sample_weight).reshape(-1, 1)
+        weighted_T = T * np.sqrt(sample_weight).reshape(-1, 1)
+        if y.ndim < 2:
+            weighted_y = y * np.sqrt(sample_weight)
+        else:
+            weighted_y = y * np.sqrt(sample_weight).reshape(-1, 1)
+        return weighted_Z, weighted_T, weighted_y
+
+    def fit(self, y, T, Z, sample_weight=None, freq_weight=None, sample_var=None):
+        """
+        Fits the model.
+
+        Parameters
+        ----------
+        y :  {(N,), (N, p)} nd array like
+            output variables
+        T :  {(N, p)} nd array like
+            treatment variables
+        Z :  {(N, p)} nd array like
+            instrumental variables
+        sample_weight : (N,) array like or None
+            Individual weights for each sample. If None, it assumes equal weight.
+        freq_weight: (N, ) array like of integers or None
+            Weight for the observation. Observation i is treated as the mean
+            outcome of freq_weight[i] independent observations.
+            When ``sample_var`` is not None, this should be provided.
+        sample_var : {(N,), (N, p)} nd array like or None
+            Variance of the outcome(s) of the original freq_weight[i] observations that were used to
+            compute the mean outcome represented by observation i.
+
+
+        Returns
+        -------
+        self : StatsModels2SLS
+        """
+        assert freq_weight is None, "freq_weight is not supported yet for this class!"
+        assert sample_var is None, "sample_var is not supported yet for this class!"
+
+        Z, T, y = self._check_input(Z, T, y, sample_weight)
+
+        self._n_out = 0 if y.ndim < 2 else y.shape[1]
+
+        # solve moment equation E[(y-theta*T)*Z]=0
+        # theta = (Z.T*T)^(-1)*Z.T*y
+        zT_t = np.dot(Z.T, T)
+        zT_y = np.dot(Z.T, y)
+        param = np.linalg.solve(zT_t, zT_y)
+        self._param = param
+
+        n_obs = y.shape[0]
+        df = len(param) if self._n_out == 0 else param.shape[0]
+
+        if n_obs <= df:
+            warnings.warn("Number of observations <= than number of parameters. Using biased variance calculation!")
+            correction = 1
+        else:
+            correction = (n_obs / (n_obs - df))
+
+        # learn cov(theta)
+        # solve first stage linear regression E[T|Z]
+        zT_z = np.dot(Z.T, Z)
+        self._thatparams = np.linalg.solve(zT_z, zT_t)
```

Comment:
okay, I will update the comments to `T̂`, hope that will help a bit. 

In terms of changing `that` to `t_hat`, I have some much longer names like `thatT_that_inv`, it will include too many `_`, I will leave it for now.